python/plugins/processing/algs/qgis/Dissolve.py

```diff
@@ -49,17 +50,24 @@ class Dissolve(GeoAlgorithm):
 
     def processAlgorithm(self, progress):
         useField = not self.getParameterValue(Dissolve.DISSOLVE_ALL)
-        fieldname = self.getParameterValue(Dissolve.FIELD)
         vlayerA = dataobjects.getObjectFromUri(
             self.getParameterValue(Dissolve.INPUT))
-        field = vlayerA.fieldNameIndex(fieldname)
         vproviderA = vlayerA.dataProvider()
-        fields = vproviderA.fields()
+        fields = QgsFields()
+        if useField:
+            fieldname = self.getParameterValue(Dissolve.FIELD)
+            fieldIdx = vlayerA.fieldNameIndex(fieldname)
+            field = vlayerA.fields()[fieldIdx]
```

Comment:
Oops right,

layer.fields().field(name) should be correct :)

On 08/31/2015 08:53 AM, bstroebl wrote:
>
> In python/plugins/processing/algs/qgis/Dissolve.py
> <https://github.com/qgis/QGIS/pull/2263#discussion_r38288013>:
>
> >          vproviderA = vlayerA.dataProvider()
> > -        fields = vproviderA.fields()
> > +        fields = QgsFields()
> > +        if useField:
> > +            fieldname = self.getParameterValue(Dissolve.FIELD)
> > +            fieldIdx = vlayerA.fieldNameIndex(fieldname)
> > +            field = vlayerA.fields()[fieldIdx]
>
> it is a QgsVectorLayer object but there is no api function "field",
> all there is is "fields" returning a QgsFields object
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/qgis/QGIS/pull/2263/files#r38288013>.
>